src/logger/message.rs

```diff
@@ -0,0 +1,111 @@
+use chrono::Local;
+use pgrx::pg_sys::PIPE_BUF;
```

Comment:
`PIPE_BUF` may be undefined according to the [source code](https://github.com/postgres/postgres/blob/8fea1bd5411b793697a4c9087c403887e050c4ac/src/include/postmaster/syslogger.h#L32), if that, `PIPE_CHUNK_SIZE`  is 512, but the `PIPE_BUF` constant in pgrx seems to be 4096 all the time. The way I write now means that `PIPE_CHUNK_SIZE` is always 4096. It is possible that the [protocol](https://github.com/postgres/postgres/blob/338fb8714c6cfe30c567ecb79812ec3879bd117f/src/backend/postmaster/syslogger.c#L948) of the log pipe is not met, because there is a sentence that says `p.len <= PIPE_MAX_PAYLOAD`
So I am a little confused about how to deal with this `PIPE_BUF`, and whether it needs to be dealt with like this
```rust
#[cfg(feature = "PIPE_BUF")]
const PIPE_BUF: usize = PIPE_BUF;
#[cfg(not(feature = "PIPE_BUF"))]
const PIPE_BUF: usize = 512;
const PIPE_CHUNK_SIZE: usize = match cfg!(feature = "PIPE_BUF") {
    true => {
        if PIPE_BUF > 65536 {
            65536
        } else {
            PIPE_BUF
        }
    }
    false => 512,
};
```

Reply:
You could use `std::mem::size_of::<PipeProtoChunk>()`.